src/thor/pathalgorithm.cc

```diff
@@ -0,0 +1,128 @@
+#include "thor/pathalgorithm.h"
+#include "baldr/datetime.h"
+#include "midgard/logging.h"
+#include "sif/edgelabel.h"
+
+#include <chrono>
+#include <ctime>
+
+namespace dt = valhalla::baldr::DateTime;
+namespace sc = std::chrono;
+
+namespace valhalla {
+namespace thor {
+
+// helper function to initialize the object from a location
+TimeInfo TimeInfo::make(valhalla::Location& location, baldr::GraphReader& reader) {
+  // No time to to track
+  if (!location.has_date_time()) {
+    return {false};
+  }
+
+  // Get the timezone of the first edge's end node that we can
```

Comment:
Incomplete comment here

Reply:
nvm misread it

Reply:
yeah it is a bit poorly worded isnt it.. seems like it could use a comma or something but my brain is too weak to figure it out.. ill just reword it completely